### ISSUE TITLE
dev-rust/cargo shouldn't depend on net-misc/curl[curl_ssl_openssl]

### ISSUE THREAD

**Issue by zetok** [Heather/gentoo-rust#129]
In case where it depends, and there is other lib satisfying ssl in curl:

```
emerge -avt cargo

These are the packages that would be merged, in reverse order:

Calculating dependencies                                   ... done!

emerge: there are no ebuilds built with USE flags to satisfy "net-misc/curl[curl_ssl_openssl]".
!!! One of the following packages is required to complete your request:
- net-misc/curl-7.42.1::gentoo (Change USE: +curl_ssl_openssl, this change violates use flag constraints defined by net-misc/curl-7.42.1: 'curl_ssl_winssl? ( elibc_Winnt ) threads? ( !adns ) ssl? ( exactly-one-of ( curl_ssl_axtls curl_ssl_gnutls curl_ssl_openssl curl_ssl_nss curl_ssl_polarssl curl_ssl_winssl ) )')
(dependency required by "dev-rust/cargo-0.2.0::rust" [ebuild])
(dependency required by "cargo" [argument])
```

**Comment by gentoo90** [comment]
#42, cargo couldn't fetch anything if built against curl without openSSL. Did you try running `cargo fetch` after installing with your patch?

**Comment by gentoo90** [comment]
Just rechecked this myself, fetch works with `curl[curl_ssl_nss]`. Looks like new curl has fixed this issue, so #130 is OK.
